main.go

```diff
@@ -107,6 +107,21 @@ func main() {
 		ExpectContinueTimeout: 1 * time.Second,
 	}
 
+	stopChan := make(chan struct{})
+	// We need this to reliably fade on DNS change, which is right
+	// now not fixed with IdleConnTimeout in the http.Transport.
+	// https://github.com/golang/go/issues/23427
+	go func() {
+		for {
+			select {
+			case <-time.After(30 * time.Second):
+				kubeConfig.Transport.CloseIdleConnections()
+			case <-stopChan:
```

Comment:
This channel is being read from twice, here and in the controller. We need two different channels or something like this, for this to work.

Reply:
see https://play.golang.org/p/M1QJsNKw4Pu we use it in skipper to shutdown >1 goroutines, too

Reply:
Ah you are right, we close the channel, I was thinking that you can't read twice!